src/GoogleFit/FitOnFhir.GoogleFit/Services/GoogleFitDataImporter.cs

```diff
@@ -63,48 +64,33 @@ public async Task Import(string userId, CancellationToken cancellationToken)
 
             if (!string.IsNullOrEmpty(tokensResponse.RefreshToken))
             {
-                _logger.LogInformation("Updating refreshToken in KV for {0}", userId);
-                await _usersKeyvaultRepository.Upsert(userId, tokensResponse.RefreshToken, cancellationToken);
+                _logger.LogInformation("Updating refreshToken in KV for {0}", googleFitId);
+                await _usersKeyvaultRepository.Upsert(googleFitId, tokensResponse.RefreshToken, cancellationToken);
             }
             else
             {
-                _logger.LogInformation("RefreshToken is empty for {0}", userId);
+                _logger.LogInformation("RefreshToken is empty for {0}", googleFitId);
             }
 
             _logger.LogInformation("Execute GoogleFitClient.DataSourcesListRequest");
             var dataSourcesList = await _googleFitClient.DataSourcesListRequest(tokensResponse.AccessToken, cancellationToken);
 
-            // Get user's info for LastSync date
-            _logger.LogInformation("Query userInfo");
-            var user = await _usersTableRepository.GetById(userId, cancellationToken);
-
-            // Generating datasetId based on event type
-            DateTimeOffset startDateDto = _utcNowFunc().AddDays(-30);
-            if (user.LastSync != null)
-            {
-                startDateDto = user.LastSync.Value;
-            }
-
-            // Convert to DateTimeOffset to so .NET unix conversion is usable
-            DateTimeOffset endDateDto = _utcNowFunc();
-
-            // .NET unix conversion only goes as small as milliseconds, multiplying to get nanoseconds
-            var startDate = startDateDto.ToUnixTimeMilliseconds() * 1000000;
-            var endDate = endDateDto.ToUnixTimeMilliseconds() * 1000000;
-            var datasetId = startDate + "-" + endDate;
-
             // Request the datasets from each datasource, based on the datasetId
             try
             {
-                await _googleFitImportService.ProcessDatasetRequests(user, dataSourcesList.DataSources, datasetId, tokensResponse, cancellationToken);
+                await _googleFitImportService.ProcessDatasetRequests(googleFitId, dataSourcesList.DataSources, tokensResponse, cancellationToken);
             }
             catch (Exception ex)
             {
                 _logger.LogError(ex, ex.Message);
             }
 
+            // Get user's info for LastSync date
+            _logger.LogInformation("Query userInfo");
+            var user = await _usersTableRepository.GetById(userId, cancellationToken);
+
             // Update LastSync column
-            user.LastSync = endDateDto;
+            user.LastSync = _utcNowFunc();
```

Comment:
I imagined that this time would represent the last occasion a sync was performed for the user by any particular platform.  It's easy enough to remove.  I thought it might still be useful.

Reply:
This would be a "LastTouched" date because it would update regardless of whether or not a sync occurred. We can keep it, but we might want to rename it.